lisa/util/process.py

```diff
@@ -17,11 +17,16 @@
 from assertpy.assertpy import AssertionBuilder, assert_that, fail
 from spur.errors import NoSuchCommandError  # type: ignore
 
-from lisa.util import LisaException, filter_ansi_escape
+from lisa.util import LisaException, RequireUserPasswordException, filter_ansi_escape
 from lisa.util.logger import Logger, LogWriter, add_handler, get_logger
 from lisa.util.perf_timer import create_timer
 from lisa.util.shell import Shell, SshShell
 
+# [sudo] password for lisatest: \r\nsudo: timed out reading password
+REQUIRE_INPUT_PASSWORD_PATTERN = re.compile(
+    r"\[sudo\] password for.+\r\nsudo: timed out reading password"
```

Comment:
How long will it timeout? Can we detect it before timeout, and raise exception earlier?

Reply:
For primekey image, the timeout is 5 minutes, we couldn't detect it before the timeout because the process is still running at that time, and calling self._process.wait_for_result() is also needed to wait. So only when it timeouts, then the process is finished and can get the stdout.